src/matchbox/server/postgresql/adapter.py

```diff
@@ -354,10 +355,17 @@ def get_dataset(self, db_schema: str, db_table: str, engine: Engine) -> Source:
                 .first()
             )
             if dataset:
+                dataset_indices: dict[str, bytes] = {}
+                for index_type, index_b64_list in dataset.indices.items():
+                    dataset_indices[index_type] = [
+                        base64.b64decode(b64.encode("utf-8")) for b64 in index_b64_list
+                    ]
                 return Source(
+                    alias=dataset.alias,
                     db_schema=dataset.schema,
                     db_table=dataset.table,
                     db_pk=dataset.id,
+                    db_columns=dataset_indices,
```

Comment:
To check my understanding: there is an assumption that the columns we use for matching are a subset of the columns indexed in the source warehouse? I think I find this 
```
db_columns = indices
```
semantically confusing